pyodide-build/pyodide_build/buildall.py

```diff
@@ -178,10 +182,11 @@ def build(self, outputdir: Path, args) -> None:
         if self.library:
             return
         if self.shared_library:
-            file_path = shutil.make_archive(
-                f"{self.name}-{self.version}", "zip", self.pkgdir / "dist"
-            )
-            shutil.copy(file_path, outputdir)
+            with self.__class__.lock:
```

Comment:
I wonder if this is the only one needed - the others really should be in separate directories and should be thread safe, I think.

Reply:
If we use absolute path for args.log_dir (so that `Path(args.log_dir)` doesn't change even when cwd changed), then I think this would be the only lock needed.

Reply:
Okay, let's start with that.

Reply:
It looks like make_archive is pretty simple and pretty easy to avoid: https://stackoverflow.com/questions/1855095/how-to-create-a-zip-archive-of-a-directory/1855118#1855118 if we wanted to go the direction of avoiding cwd changes in the first place.

Reply:
Thanks for the investigation! I think the two solutions are all temporary and the long term solution would be moving some stuff in `buildall` to `buildpkg`, so for now let's go with this locking-based one and revisit this later.

Reply:
Yes, I think #2256 is the next priority. After that, I'll probably look into some refactoring of buildall/buildpkg related to #2246 - if this was async, it wouldn't matter, and if it was moved to buildpkg, it also wouldn't matter. I'm also interested in working on supporting CMake packages, but that also requires #2256 (or at least I'd rather learn the system after the rewrite and not before).

Reply:
> Yes, I think #2256 is the next priority.

Yes, #2256 will be a big change in our build system, and I will be glad if you can help us review it.